sros2/sros2/api/__init__.py

```diff
@@ -189,12 +214,21 @@ def create_ecdsa_param_file(path):
     run_shell_command('%s ecparam -name prime256v1 > %s' % (openssl_executable, path))
 
 
-def create_ca_key_cert(ecdsa_param_path, ca_conf_path, ca_key_path, ca_cert_path):
-    openssl_executable = find_openssl_executable()
-    check_openssl_version(openssl_executable)
-    run_shell_command(
-        '%s req -nodes -x509 -days 3650 -newkey ec:%s -keyout %s -out %s -config %s' %
-        (openssl_executable, ecdsa_param_path, ca_key_path, ca_cert_path, ca_conf_path))
+def create_ca_key_cert(ca_key_out_path, ca_cert_out_path):
+    private_key = ec.generate_private_key(ec.SECP256R1, cryptography_backend())
```

Comment:
Can you add a comment regarding the use of `SECP256R1`? Why are we using that one?

Reply:
I'm not 100% sure on the choice, it's the internal name for what we are specifying in this call  https://github.com/ros2/sros2/pull/126/files#diff-4db4152194feafaa6ef7a7fd1c647c53R214 with the arg "prime256v1". @ruffsl maybe you can shed more light on the choice of curve?

Reply:
> can shed more light on the choice of curve?

The choice of the curve `prime256v1` is specified by the [DDS-Security specification](https://www.omg.org/spec/DDS-SECURITY/1.1/PDF) in section 9.3.1.
> The public key of the CA shall be either a 2048-bit RSA key [44] or else a 256-bit Elliptic Curve Key for the prime256v1 curve [41], also known as the NIST P-256 curve [42].

`SECP256R1` is another alias for that same curve: https://tools.ietf.org/search/rfc4492#appendix-A so this is the right change :+1: 

> Can you add a comment regarding the use of SECP256R1?

:+1: for adding a comment for future code readers

Reply:
Thank you @mikaelarguedas, that's exactly the ammo required for a good comment!

Reply:
comment added